File: hw6/Shape.java
package hw6;

import java.io.*;
import java.util.*;

public abstract class Shape{
    protected String color;
    public Shape(String inp){
        color = inp;
    }
    public abstract double get_area();
    public void print(){
        //
    }
    public String get_color(){
        return color;
    }
}

File: hw6/Shape.cpp
#include "Shape.h"

Shape::Shape(const string& inp){
    color = inp;
    cout << "Shape construct." << endl;
}

Shape::~Shape(){
    cout << "Shape destruct." << endl;
}

void Shape::print(){
    //
}

string Shape::getcolor(){
    return color;
}